MBBSEmu/DOS/Interrupts/Int21h.cs

```diff
@@ -564,7 +596,83 @@ private void GetDeviceInformation()
             }
         }
 
-        private void OpenFile()
+        private void GetOrPutFileAttributes_0x43()
+        {
+            /*
+            INT 21 - AH = 43h DOS 2+ - GET/PUT FILE ATTRIBUTES (CHMOD)
+            AL =
+                0 get file attributes
+                1 put file attributes
+            CX = file attribute bits
+                0 = read only
+                1 = hidden file
+                2 = system file
+                3 = volume label
+                4 = subdirectory
+                5 = written since backup
+                8 = shareable (Novell NetWare)
+            DS:DX -> ASCIZ file name
+            Return: CF set on error
+                AX = error code
+                CX = file attributes on get
+            */
+            var file = Encoding.ASCII.GetString(_memory.GetString(_registers.DS, _registers.DX, stripNull: true));
+            if (_registers.AL != 0)
+                throw new NotImplementedException();
+
+            FileInfo fileInfo = new FileInfo(file);
+            if (!fileInfo.Exists)
+            {
+                _registers.F = _registers.F.SetFlag((ushort)EnumFlags.CF);
+                _registers.AX = (ushort)DOSErrorCode.FILE_NOT_FOUND;
+                return;
+            }
+
+            _registers.CX = 0;
+            if (fileInfo.IsReadOnly)
+                _registers.CX |= (ushort)EnumDirectoryAttributeFlags.ReadOnly;
+            if (fileInfo.Attributes.HasFlag(FileAttributes.Hidden))
+                _registers.CX |= (ushort)EnumDirectoryAttributeFlags.Hidden;
+            if (fileInfo.Attributes.HasFlag(FileAttributes.System))
+                _registers.CX |= (ushort)EnumDirectoryAttributeFlags.System;
+            if (fileInfo.Attributes.HasFlag(FileAttributes.Directory))
+                _registers.CX |= (ushort)EnumDirectoryAttributeFlags.Directory;
+            if (fileInfo.Attributes.HasFlag(FileAttributes.Archive))
+                _registers.CX |= (ushort)EnumDirectoryAttributeFlags.Archive;
+
+            _registers.F = _registers.F.ClearFlag((ushort)EnumFlags.CF);
+            return;
+        }
+
+        private void CloseFile_0x3E()
+        {
+            /*
+            INT 21 - AH = 3Eh DOS 2+ - CLOSE A FILE WITH HANDLE
+            BX = file handle
+            Return: CF set on error
+                AX = error code
+            */
+            switch (_registers.BX)
```

Comment:
Could do the enum cast here, don't think it matters performance wise. Would just be less code :P

Reply:
Leaving in for now since next PR implements this fully